cmd/kube-egress-gateway-cnimanager/cmd/serve.go

```diff
@@ -85,14 +87,25 @@ func ServiceLauncher(cmd *cobra.Command, args []string) {
 	logger.SetDefaultLogger(zapr.NewLogger(zapLog))
 	logger := logger.GetLogger()
 
-	k8sClient := startKubeClient(ctx, logger)
+	k8sCluster := getKubeCluster(ctx, logger)
```

Comment:
I'm wondering if creating a new controller would be a better option.
https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg@v0.19.0#hdr-EventHandler
https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg@v0.19.0#hdr-Watching_and_EventHandling

Reply:
I considered this option, but I choose to use eventHandler directly because our change is simple, we just need to monitor the node taint and I feel creating a controller is too much here. Any reason using a controller might be better?

Reply:
It will deduplicate the event